services/registration/src/routes/application.ts

```diff
@@ -224,6 +225,26 @@ applicationRouter.route("/:id/actions/submit-application").post(
       throw new BadRequestError("Unable to submit application data. Branch is not currently open.");
     }
 
+    const applicantType = branch.applicationGroup;
+    const autoConfirm = branch.automaticConfirmation;
+
+    if (
+      autoConfirm?.enabled &&
+      (applicantType === "PARTICIPANT" || existingApplication.email in autoConfirm.emails!)
+    ) {
+      await ApplicationModel.findByIdAndUpdate(
+        req.params.id,
+        {
```

Comment:
Also, this update occurs too early, and you're not validating any of the application data as seen later in the handler. This checks will need to occur much later in the handler